debezium-core/src/main/java/io/debezium/relational/topic/TopicMappingProvider.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.relational.topic;
+
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+
+import io.debezium.relational.Table;
+
+/**
+ * @author Randall Hauch
+ *
+ */
+public interface TopicMappingProvider {
+
+    /**
+     * A mapping strategy for a table.
+     */
+    public static interface TopicMapping {
+        /**
+         * Get the name of the topic given for the table.
+         * 
+         * @return the topic name; may be null if this strategy could not be applied
+         */
+        String getTopicName();
+
+        /**
+         * Get the name of the schema for the key.
+         * 
+         * @return the key's schema name; may not be null
+         */
+        default String getKeySchemaName() {
```

Comment:
Why are we providing a way to customize the KeySchemaName and ValueSchemaName? As I understand it, the motivation for this PR was custom DML topic mappings. That doesn't seem to require customizing the KeySchemaName or ValueSchemaName...

That said, I guess it doesn't hurt to add that customization feature.